src/box/applier.cc

```diff
@@ -699,13 +699,12 @@ applier_register(struct applier *applier, bool was_anon)
 	/* Send REGISTER request */
 	struct iostream *io = &applier->io;
 	struct xrow_header row;
-
 	memset(&row, 0, sizeof(row));
-	/*
-	 * Send this instance's current vclock together
-	 * with REGISTER request.
-	 */
-	xrow_encode_register(&row, &INSTANCE_UUID, box_vclock);
+	struct register_request req;
+	memset(&req, 0, sizeof(req));
```

Comment:
You already do a memset inside `xrow_encode_replication_request`. Let's not duplicate it here and in other places below.

Reply:
I prefer to keep initialization with zeros. I think structs always have to be initialized either via memset or via `= {.field1 = ..., .field2 = ..., ...}`. Especially if they are used as in-parameters to a function (so the function expects them to be fully valid and initialized). Right now all fields are initialized explicitly indeed, but if a new field would be added, it wouldn't be so. You would need to walk all the usage places and initialize it with 0 explicitly if it is optional or with some meaningful value. If you forget to do so (code will compile anyway), you will get garbage in this field which will occasionally lead to failures.

Reply:
> Is it the row memset you're referring to?

Yep. I got confused here.
Anyway, there is an extraneous `memset` 2 lines above. Not in other places though.

Reply:
Thanks, done.